src/main/connections/BaseConnection.ts

```diff
@@ -0,0 +1,15 @@
+import { SqlModificationResult, SqlQueryResult } from '@nodescript/adapter-sql-protocol';
+import { Logger } from '@nodescript/logger';
+import { dep } from 'mesh-ioc';
+import { PoolConnection } from 'mysql2/promise';
+import { PoolClient } from 'pg';
+
+export abstract class BaseConnection {
+    @dep() logger!: Logger;
+
+    constructor(protected client: PoolClient | PoolConnection) { this.client = client; }
```

Comment:
I think `this.client = client` part is not actually required, since the member is defined as `protected` in args — so an empty constructor body will do.